lib/symboldatabase.cpp

```diff
@@ -171,7 +171,7 @@ void SymbolDatabase::createSymbolDatabaseFindAllScopes()
                     if (!new_type) {
                         typeList.push_back(Type(new_scope->classDef, new_scope, scope));
                         new_type = &typeList.back();
-                        scope->definedTypes.push_back(new_type);
+                        scope->definedTypesMap[new_type->name()] = new_type;
                     } else
```

Comment:
Since the key of the map (i.e. ```new_type->name()```) is contained in the value of the map (i.e. ```new_type```) the usage of a set instead of a map would be totally sufficient at this place.